wgpu-types/src/lib.rs

```diff
@@ -232,6 +234,64 @@ bitflags::bitflags! {
         ///
         /// This is a web and native feature.
         const DEPTH_CLIP_CONTROL = 1 << 0;
+        /// Enables use of Timestamp Queries. These queries tell the current gpu timestamp when
+        /// all work before the query is finished. Call [`CommandEncoder::write_timestamp`],
+        /// [`RenderPassEncoder::write_timestamp`], or [`ComputePassEncoder::write_timestamp`] to
+        /// write out a timestamp.
+        ///
+        /// They must be resolved using [`CommandEncoder::resolve_query_sets`] into a buffer,
+        /// then the result must be multiplied by the timestamp period [`Queue::get_timestamp_period`]
+        /// to get the timestamp in nanoseconds. Multiple timestamps can then be diffed to get the
+        /// time for operations between them to finish.
+        ///
+        /// Supported Platforms:
+        /// - Vulkan
+        /// - DX12
+        ///
+        /// This is currently unimplemented on Metal.
+        ///
+        /// This is a web and native feature.
+        const TIMESTAMP_QUERY = 1 << 1;
+        /// Allows non-zero value for the "first instance" in indirect draw calls.
+        ///
+        /// Supported Platforms:
+        /// - Vulkan (mostly)
+        /// - DX12
+        /// - Metal
+        ///
+        /// This is a web and native feature.
+        const INDIRECT_FIRST_INSTANCE = 1 << 2;
+
+        // 3..8 available
+
+        // Shader:
+
+        /// Allows shaders to acquire the FP16 ability
+        ///
+        /// Note: this is not supported in naga yet，only through spir-v passthrough right now.
+        ///
+        /// Supported Platforms:
+        /// - Vulkan
+        /// - Metal
+        ///
+        /// This is a web and native feature.
+        const SHADER_F16 = 1 << 8;
+
+        // 9..14 available
+
+        // Texture Formats:
+
+        // ? const FORMATS_TIER_1 = 1 << 14; (https://github.com/gpuweb/gpuweb/issues/3837)
+        // ? const RW_STORAGE_TEXTURE_TIER_1 = 1 << 15; (https://github.com/gpuweb/gpuweb/issues/3838)
+        // TODO const BGRA8UNORM_STORAGE = 1 << 16;
+        // ? const NORM16_FILTERABLE = 1 << 17; (https://github.com/gpuweb/gpuweb/issues/3839)
+        // ? const NORM16_RESOLVE = 1 << 18; (https://github.com/gpuweb/gpuweb/issues/3839)
+        // TODO const FLOAT32_FILTERABLE = 1 << 19;
+        // ? const FLOAT32_BLENDABLE = 1 << 20; (https://github.com/gpuweb/gpuweb/issues/3556)
+        // ? const 32BIT_FORMAT_MULTISAMPLE = 1 << 21; (https://github.com/gpuweb/gpuweb/issues/3844)
+        // ? const 32BIT_FORMAT_RESOLVE = 1 << 22; (https://github.com/gpuweb/gpuweb/issues/3844)
+        // TODO const RG11B10UFLOAT_RENDERABLE = 1 << 23;
```

Comment:
Could you clarify a bit what all this is

Reply:
The `TODO`'s are for features we are missing from the spec.
The features starting with a `?` are features that might become part of the spec or at the very least we can implement as native features. I researched all the formats and capabilities that we can expose and grouped them by hardware support.

Reply:
That's great!

Could you put this explanation in the code for future reference?